File: 04 Move Zeroes/Move Zeroes.cpp
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int nonZero = 0, index = 0, flag = 0;
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] != 0)
                nonZero++;
        }
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] != 0 && flag == 0)
            {
                nums[index] = nums[i];
                if (i >= nonZero)
                    nums[i] = 0;
                index++;
                if (index == nonZero)
                    flag = 1;
            }
            else
                nums[i] = 0;
        }
    }
};

File: 04 Move Zeroes/Move Zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
    let cnt = 0;
    for (let i = nums.length - 1; i >= 0; i--) {
      if (nums[i] == 0) {
        cnt++;
        nums.splice(i, 1);
      }
    }
    while (cnt--)
      nums.push(0);
    return nums;
  
  };